kernel/kernel.h

```diff
@@ -65,7 +65,6 @@ void _abort(const char *, const char *, const char *)
 
 /* cpu_<architecture>.c: low-level CPU functions */
 void cpu_init(void);
-void cpu_halt(void) __attribute__((noreturn));
```

Comment:
Nit: Can we please keep the arch-dependent `cpu_halt()` implementations and call those from the muen and virtio `platform_exit()`? There should not be any arch-dependent inline `asm` outside of the `cpu_*` modules.

Reply:
Added back. The only change now is that `cpu_halt` for ARM is not using the ukvm hypercall anymore. I "borrowed" the freebsd cpu_halt implementation from arm64. Keep in mind that `cpu_halt` is only used from virtio, and we have never tested virtio on arm64.